src/abstract-sql-compiler.ts

```diff
@@ -869,10 +882,47 @@ ${compileRule(definitionAbstractSql as AbstractSqlQuery, engine, true).replace(
 		const depends: string[] = [];
 		const createSqlElements: string[] = [];
 		const createIndexes: string[] = [];
+		const createComputedFunctions: string[] = [];
+		const dropComputedFunctions: string[] = [];
 
 		for (const field of table.fields) {
 			const { fieldName, references, dataType, computed } = field;
-			if (!computed) {
+			if (computed != null) {
+				if (!Array.isArray(computed) && computed !== true) {
+					let sqlType =
+						sbvrTypes[dataType as keyof typeof sbvrTypes]?.types[engine];
+					if (sqlType == null) {
+						throw new Error(
+							`Unknown data type '${dataType}' for engine: ${engine}`,
+						);
+					}
+					if (typeof sqlType === 'function') {
+						sqlType = sqlType.castType;
+					}
+					const fnSignature = `"computed_${table.name}_${fieldName}"("${table.name}" "${table.name}")`;
```

Comment:
Should we trim function names like we do for triggers with `slice(0, 63)`?

Reply:
Yeah I've trimmed it, along with shrinking the prefix and allowing a name of choice for the function to be specified (in case it becomes necessary, and also avoiding copy/pasting the fn name construction in multiple places)